### PR TITLE
Testing ; Round 1

### LINKED ISSUE #26: Unit Testing ; Welcome Page
- [ ] Test that page renders
- [ ] Test there is a button to navigate to `Lobby`

### LINKED ISSUE #27: Unit Testing ; Reflection Form
- [ ] Test for whole page rendering
- [ ] Test to make sure `textArea` renders
- [ ] Test for presence of buttons
- [ ] Test for function firing when clicked

### PR DESCRIPTION
**Pull request by BrigetteDoelp**
#### Description
- Add unit tests for 80% of components.
- Add integration testing for 80% of components.
#### Motivation and Context
Apart from a robust testing suite being in the project spec, we wanted to practice with writing them and setting good `TDD` habits.
#### How Has This Been Tested?
- [x] In console
- [ ] In browser
- [x] With testing files
- [ ] Other Please specify:
#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Styling / Design
- [x] Testing
- [ ] Refactor
#### Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
#### Issues Addressed: 
Resolves #27 
Resolves #26

### CODE REVIEW

File: src/Components/ReflectionLibrary/ReflectionLibrary.test.js [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
-// It should display ReflectionCards after a successful fetch request
-// It should display an error message after an unsuccessful fetch request
+import { render, waitFor, screen } from '@testing-library/react';
+import ReflectionLibrary from './ReflectionLibrary';
+import { MemoryRouter } from 'react-router-dom';
+import { getReflections } from '../../apiCalls.js';
+import '@testing-library/jest-dom';
+
+jest.mock('../../apiCalls.js');
+
```
Comment (Kristiannmiller) [inline src/Components/ReflectionLibrary/ReflectionLibrary.test.js:8]: Did we try jest.mock(getReflections) here?

File: src/Components/App/App.test.js [hunk 2]
```diff
@@ -1,10 +1,31 @@
...
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
-});
 
-// A user sees the quote of the day and the emptyLogo
+describe('App', ()=>{
+  describe('Unit Tests', ()=>{
```
Comment (nathanielmillard) [inline src/Components/App/App.test.js:6]: Excited to get into these integration tests, when we get there.

File: src/Components/Lobby/Lobby.js [hunk 3]
```diff
@@ -22,30 +22,33 @@ class Lobby extends Component {
 
     return (
       <section className='Lobby'>
-        <h1>Welcome Back, it's {new Date().toDateString()}</h1>
-        <h2>How are you feeling today?</h2>
-        <section className='moodContainer'>
+        <h1 data-testid='welcomeMessage'>Welcome Back, it's {new Date().toDateString()}</h1>
+        <h2 data-testid='checkInQuestion'>How are you feeling today?</h2>
```
Comment (nathanielmillard) [inline src/Components/Lobby/Lobby.js:26]: thanks for putting testIds back in!

File: src/Components/Lobby/Lobby.test.js [hunk 4]
```diff
@@ -1,14 +1,75 @@
...
+      const linkToFeeling = screen.getByTestId('feel');
+      const linkToBreathing = screen.getByTestId('breathe');
+      expect(moodButtonContainer).toBeInTheDocument();
+      expect(linkToReflecting).toBeInTheDocument();
+      expect(linkToFeeling).toBeInTheDocument();
+      expect(linkToBreathing).toBeInTheDocument();
+    });
+  })
+
+  describe('Integration Tests', () => {
```
Comment (nathanielmillard) [inline src/Components/Lobby/Lobby.test.js:44]: Excited to get into more inegration tests, these ones look great :)

File: src/Components/ReflectionLibrary/ReflectionLibrary.test.js (hunk 1 above)
Comment (nathanielmillard) [reply src/Components/ReflectionLibrary/ReflectionLibrary.test.js:8]: Not here, but we did in test itself and it was still buggy. But could have been a syntax mistake on our part.

File: src/Components/FeelingRoom/fineAnimations.scss [hunk 5]
```diff
@@ -0,0 +1,67 @@
+
```
Comment (nathanielmillard) [inline src/Components/FeelingRoom/fineAnimations.scss:1]: Strange to see this show up here too, but shouldn't impact functionality or impact our marge.

**Review by Kristiannmiller — APPROVED** [review]
Testing is looking good! I think we should definitely get back together as a group to do some of the more intensive integration tests in the next testing branch. It's good practice and I think we could all use a little more exposure to tricky testing situations

**Review by nathanielmillard — APPROVED** [review]
Give that we were on a call through the merging of this, all of our conserns were adressed in live time. Over all I would say things are looking good!
Exciting to see the number of test file we have up and running, and getting deeper into integration tests, user events, and mocking functions is going to be valuable before wednesday.
Great work!

**Pull request merged**